### ISSUE TITLE
Gradle plugin is failing to install in ATH

### ISSUE THREAD

**Issue by timja** [jenkinsci/acceptance-test-harness#985]
```
java.net.SocketException: Connection reset
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:186)
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:140)
	at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:252)
	at java.base/java.io.BufferedInputStream.read1(BufferedInputStream.java:292)
	at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:351)
	at java.base/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:788)
	at java.base/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:723)
	at java.base/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:747)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1620)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1525)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3129)
	at java.base/java.net.URLConnection.getHeaderFieldLong(URLConnection.java:636)
	at java.base/java.net.URLConnection.getContentLengthLong(URLConnection.java:508)
	at java.base/java.net.URLConnection.getContentLength(URLConnection.java:492)
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1277)
Caused: java.net.SocketException: Connection reset
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1989)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1984)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1983)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1541)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1525)
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1296)
Caused: java.io.IOException: Failed to load http://0.0.0.0:56148/gradle.hpi to /Users/timja/code/jenkins/acceptance-test-harness/target/jenkinshome15427209656703364098/plugins/gradle.jpi.tmp
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1307)
Caused: java.io.IOException: Failed to download from http://0.0.0.0:56148/gradle.hpi
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1341)
	at hudson.model.UpdateCenter$DownloadJob._run(UpdateCenter.java:1898)
	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:2210)
	at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1872)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:121)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

Causes all Gradle plugin tests to fail and also any that depend on the plugin e.g. artifactory plugin

**Comment by timja** [comment]
I was able to install it fine in Jenkins 2.376 from update center

**Comment by timja** [comment]
Gradle plugin version 2.0 was released 7 days ago, likely the cause

**Comment by timja** [comment]
Maybe:
```
master00000|2022-11-02 22:47:32.452+0000 [id=88]	INFO	h.model.UpdateCenter$DownloadJob#run: Starting the installation of gradle on behalf of anonymous
Downloading: https://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/gradle/2.0/gradle-2.0.hpi
org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.jenkins-ci.plugins:gradle:hpi:2.0 in repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/)
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:39)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:355)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:67)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:581)
```

**Comment by timja** [comment]
https://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/gradle/1.40/

vs

https://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/gradle/2.0/

**Comment by timja** [comment]
@jglick any thoughts on whether this is a valid change for the plugin to change its extension?

We can probably hack in a `if gradle extension = jpi...`

**Comment by timja** [comment]
I tested including the dependency on gradle in jenkins-infra-test-plugin.

`mvn install` works, 

`mvn maven-hpi-plugin:resolve-test-dependencies` does resolve jpi fine

```
[INFO] --- maven-hpi-plugin:3.27:resolve-test-dependencies (default-resolve-test-dependencies) @ jenkins-infra-test ---
Downloading from repo.jenkins-ci.org: https://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/gradle/2.0/gradle-2.0.jpi
Downloaded from repo.jenkins-ci.org: https://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/gradle/2.0/gradle-2.0.jpi (21 MB at 9.0 MB/s)
```

but `mvn hpi:run` fails

```
[ERROR] Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.27:run (default-cli) on project jenkins-infra-test: Unable to copy dependency plugin: Could not find artifact org.jenkins-ci.plugins:gradle:hpi:2.0 in repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/) -> [Help 1]
```

**Comment by jglick** [comment]
> would maven-hpi-plugin handle this and allow depending on a plugin with a jpi extension

I guess. `.hpi` is historical but the transition to `.jpi` is incomplete: while `PluginManager` prefers `.jpi`, Maven-based tooling still generates `.hpi`.

**Comment by timja** [comment]
They've accepted my PR to revert it https://github.com/jenkinsci/gradle-plugin/pull/211

**Comment by c00ler** [comment]
@timja We've released a new version of the Gradle plugin `2.1` that uses `hpi` packaging format: https://repo.jenkins-ci.org/artifactory/public/org/jenkins-ci/plugins/gradle/2.1/. Thank you for reporting the issue and contributing the fix!

**Comment by timja** [comment]
Thanks!

Confirmed fixed
